spring-kafka/src/main/java/org/springframework/kafka/core/KafkaTemplate.java

```diff
@@ -456,6 +456,15 @@ public void setObservationConvention(KafkaTemplateObservationConvention observat
 		this.observationConvention = observationConvention;
 	}
 
+	/**
+	 * Configure the {@link ObservationRegistry} to use for recording observations.
+	 * @param observationRegistry the observation registry to use.
+	 * @since 3.3
+	 */
+	public void setObservationRegistry(ObservationRegistry observationRegistry) {
+		this.observationRegistry = observationRegistry;
```

Comment:
`Assert.notNull(observationRegistry, "'observationRegistry' must not be null");`